csharp/HOLMS.Platform/HOLMS.Platform/Support/Currency/DollarCents.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using HOLMS.Types.Primitive;
+
+namespace HOLMS.Platform.Support.Currency {
+    public struct DollarCents : IComparable<DollarCents>, IEquatable<DollarCents> {
+        private const uint OneMillion = 1000000;
+        public readonly bool IsNegative;
+        public readonly uint Dollars;
+        public readonly uint Cents;
+
+        public DollarCents(bool isNegative, uint dollars, uint cents) {
+            IsNegative = isNegative;
+            Dollars = dollars;
+            Cents = cents;
+        }
+
+        public DollarCents(MonetaryAmount amt) {
+            var dc = FromCents((int) Math.Round(amt.Microdollars / (decimal) OneMillion, MidpointRounding.ToEven));
```

Comment:
I believe this is wrong. Your conversion math with units is:

`n(microdollars) * 10**-6 ( dollars per microdollar) == n-prime (dollars)`

But you're passing that number into the "From Cents" factory which would mean that your conversion factor of microdollars to dollars is misapplied.